azurerm/internal/services/managementgroup/data_source_management_group.go

```diff
@@ -21,8 +22,9 @@ func dataSourceArmManagementGroup() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"group_id": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ManagementGroupName,
```

Comment:
Here group_id means this one in its ID:
```
/providers/Microsoft.ManagementGroup/managementGroups/{group_id}
```
the thing that in other resources' IDs at the same position would be called as a `name`, for instance
```
/subscriptions/{subsID}/resourceGroups/group1/providers/Microsoft.Compute/virtualMachines/{name}
```
And another reason is that if we call this `group_id` in the validation function, the validation function would looks like `ManagementGroupGroupID` or `ManagementGroupID`. This would be confusing with the validation function of the real management group ID. Therefore I took this name.